platinum-sw-register.html

```diff
@@ -376,8 +374,26 @@
     },
 
     attached: function() {
+      var isSupported = 'serviceWorker' in navigator;
+
+      if (!isSupported) {
+        this._setState('unsupported');
+        return;
+      }
+
       if (this.autoRegister) {
-        this.async(this.register);
+        if (document.readyState !== 'complete' &&
+            document.readyState !== 'interactive') {
+
+          document.onreadystatechange = function() {
```

Comment:
@sorvell claims that the children in the light dom should be upgraded after `document.readyState === 'interactive'` plus a rAF assuming all the light dom webcomponents are html-imported synchronously.

The polymer 1 version of this assumed that they were all already upgraded and would simply go through them and see if they had a function, called it, and then shoved the concatenated results into a SW.

Polymer 2 upgrade / lifecycle order doesn't work in the same way